src/main/kotlin/lotto/domain/Result.kt

```diff
@@ -1,19 +1,19 @@
 package lotto.domain
 
-class Result(matchHistories: List<Int>) {
-    val values: Map<Int, Int>
+class Result(matchHistories: List<MatchingCount>) {
+    val values: Map<Rank, Int>
 
     init {
-        val map = mutableMapOf(3 to 0, 4 to 0, 5 to 0, 6 to 0)
-        matchHistories.forEach { map[it] = map.getOrDefault(it, 0) + 1 }
-        this.values = map
+        this.values = matchHistories.fold(mutableMapOf()) { map, matchingCount ->
+            map[Rank.valueOf(matchingCount)] = map.getOrDefault(Rank.valueOf(matchingCount), 0) + 1
+            map
+        }
+
+        Rank.values().forEach { rank ->
+            values[rank] = values[rank]?.let { values[rank] } ?: 0
+        }
     }
 
-    fun calculateProfit(money: Money): Double = values.map { prizeMoney[it.key]?.times(it.value) ?: 0 }
+    fun calculateProfit(money: Money): Double = values.map { it.key.winningMoney * it.value }
```

Comment:
[map](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map.html)의 설명 맨밑의 예시와 같이 key,value가 무엇을 의미하는지 적으면 더 직관적일 것 같아요~~

```suggestion
   fun calculateProfit(money: Money): Double = values.map { (rank, matchCount) -> rank.winningMoney * matchCount }
```